pycqed/instrument_drivers/physical_instruments/ZurichInstruments/UHFQuantumController.py

```diff
@@ -295,293 +435,190 @@ def load_default_settings(self, upload_sequence=True):
-        self._daq.setInt('/' + self._device + '/quex/rl/readout', 0)
-        # It would be better to move this call in to the initialization function
-        # in order to save time here
-        enable_path = '/' + self._device + '/awgs/0/enable'
-        self._daq.subscribe(enable_path)
-
-        # Added for testing purposes, remove again according to how the AWG is
-        # started
-        self._daq.setInt('/' + self._device + '/awgs/0/single', 1)
-        self._daq.setInt(enable_path, 1)
-
-        # Wait for the AWG to finish
-        gotit = False
-        accumulated_time = 0
-        while not gotit and accumulated_time < timeout:
-            dataset = self._daq.poll(acquisition_time, 1, 4, True)
-            if enable_path in dataset and dataset[enable_path]['value'][0] == 0:
-                gotit = True
-            else:
-                accumulated_time += acquisition_time
-
-        if not gotit:
-            print("Error: AWG did not finish in time!")
-            return None
-
-        # Acquire data
-        gotem = [False]*len(channels)
-        for n, c in enumerate(channels):
-            p = paths[c]
-            dataset = self._daq.get(p, True, 0)
-            if p in dataset:
-                for v in dataset[p]:
-                    data[c] = np.concatenate((data[c], v['vector']))
-                if len(data[c]) >= samples:
-                    gotem[n] = True
+                path = self._get_full_path('qas/0/monitor/inputs/{}/wave'.format(c))
+                self._acquisition_nodes.append(path)
+                self.subs(path)
+            # Enable automatic readout
+            self.qas_0_monitor_reset(1)
+            self.qas_0_monitor_enable(1)
+            self.qas_0_monitor_length(samples)
+            self.qas_0_monitor_averages(averages)
+            ro_mode = 1
+
+        self.set('awgs_0_userregs_{}'.format(UHFQC.USER_REG_LOOP_CNT), loop_cnt)     
+        self.set('awgs_0_userregs_{}'.format(UHFQC.USER_REG_RO_MODE), ro_mode)     
+        self.set('awgs_0_userregs_{}'.format(UHFQC.USER_REG_AVG_CNT), averages)
+        if self.wait_dly() > 0 and not self._awg_program_features['wait_dly']:
+            raise ziUHFQCSeqCError('Trying to use a delay of {} using an AWG program that does not use \'wait_dly\'.'.format(self.wait_dly()))
+        self.set('awgs_0_userregs_{}'.format(UHFQC.USER_REG_WAIT_DLY), self.wait_dly())
+        self.subs(self._get_full_path('auxins/0/sample'))
 
-        if not all(gotem):
-            print("Error: Didn't get all results!")
-            for n, c in enumerate(channels):
-                print("    : Channel {}: Got {} of {} samples",
-                      c, len(data[c]), samples)
-            return None
+        # Generate more dummy data
+        self.auxins_0_averaging(8)
 
-        # print("data type {}".format(type(data)))
-        return data
+    def acquisition_arm(self, single=True) -> None:
+        # time.sleep(0.01)
+        self.awgs_0_single(single)
```

Comment:
@AdriaanRol 
There is just one AWG on the UHF. The nodes are organized in a similar fashion as on the HDAWG. Therefore, we need to use the '0' index here.

'single' means to activate the AWG in single-shot mode. It will run the program once and then stop. If 'single' is not set, then the AWG will restart the program once it finishes, which will greatly screw with the measurements.